src/reducers/entities/channel_categories.ts

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {combineReducers} from 'redux';
+
+import {CategoryTypes} from '../../constants/channel_categories';
+
+import {ChannelCategoryTypes, TeamTypes, UserTypes} from 'action_types';
+
+import {GenericAction} from 'types/actions';
+import {ChannelCategory} from 'types/channel_categories';
+import {Team, TeamMembership} from 'types/teams';
+import {$ID, IDMappedObjects, RelationOneToOne} from 'types/utilities';
+
+export function byId(state: IDMappedObjects<ChannelCategory> = {}, action: GenericAction) {
+    switch (action.type) {
+    case TeamTypes.RECEIVED_MY_TEAM_MEMBER: {
+        // This will be removed once categories are sent by the server
+        const member: TeamMembership = action.data;
+
+        // Note that this adds new categories before state to prevent overwriting existing categories
+        return {
+            ...makeDefaultCategories(member.team_id),
+            ...state,
+        };
+    }
+    case TeamTypes.RECEIVED_MY_TEAM_MEMBERS: {
+        // This will be removed once categories are sent by the server
+        const members: TeamMembership[] = action.data;
+
+        return members.reduce((nextState, member) => {
+            // Note that this adds new categories before state to prevent overwriting existing categories
+            return {
+                ...makeDefaultCategories(member.team_id),
+                ...nextState,
+            };
+        }, state);
+    }
+
+    // case ChannelCategoryTypes.RECEIVED_CATEGORIES: {
```

Comment:
Can we remove this code? Or are we keeping it for Phase 2?

Reply:
We'll keep it for phase 2 for now